static/app/components/searchSyntax/parser.tsx

```diff
@@ -455,12 +555,95 @@ export type ParseResult = Array<
   | TokenResult<Token.Spaces>
 >;
 
+/**
+ * Configures behavior of search parsing
+ */
+export type SearchConfig = {
+  /**
+   * Keys which are considered valid for duration filters
+   */
+  durationKeys: Set<string>;
+  /**
+   * Text filter keys we allow to have operators
+   */
+  textOperatorKeys: Set<string>;
+  /**
+   * Keys considered valid for the percentage aggregate and may have percentage
+   * search values
+   */
+  percentageKeys: Set<string>;
+  /**
+   * Keys considered valid for numeric filter types
+   */
+  numericKeys: Set<string>;
+  /**
+   * Keys considered valid for date filter types
+   */
+  dateKeys: Set<string>;
+  /**
+   * Keys considered valid for boolean filter types
+   */
+  booleanKeys: Set<string>;
+  /**
+   * Enables boolean filtering (AND / OR)
+   */
+  allowBoolean: boolean;
+};
+
+const defaultConfig: SearchConfig = {
```

Comment:
@Zylphrex @wmak I know yall showed me somewhere else I can get these, but I hardcoded them the same way we do on the backend for now. What do yall think, should we try and unify?

Reply:
In quite a few places we actually want the reverse mapping from name to the type so this may not be ideal. Though we could just generate that reverse map.